bin/anthology/index.py

```diff
@@ -39,6 +39,24 @@ def load_stopwords(language):
     return [t for w in get_stop_words(language) for t in slugify(w).split("-")]
 
 
+def score_variant(name):
+    """Heuristically assign scores to names, with the idea of assigning higher
+    scores to spellings more likely to be the correct canonical variant."""
+    name = repr(name)
+    # Prefer longer variants
+    score = len(name)
+    # Prefer variants with non-ASCII characters
+    score += sum((ord(c) > 127) for c in name)
+    # Penalize upper-case characters after word boundaries
+    score -= sum(any(c.isupper() for c in w[1:]) for w in re.split(r"\W+", name))
```

Comment:
Why this rule and the next?

Reply:
I think it's in order to downweight full-caps and all lower case.  I copied this routine from find_name_variants.py

Reply:
Hm, but it would also prefer "Van Noord" also "van Noord" and "Dusell" over "DuSell", right? @mbollmann

Reply:
Sure.  It also prefers "Van Durme" over "van Durme".  This can be overridden by editing name_variants.yaml, or by editing the xml. I think we said earlier that the xml should reflect how we want the paper to be cited, which does not always exactly match the pdf.  If the person's name is really DuSell and the pdf says Dusell, we should correct the xml.

Reply:
I guess we can revisit it later if we see any mistakes!